pkg/dns/hostresolv.go

```diff
@@ -0,0 +1,171 @@
+package dns
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	gonet "net"
+	"sync"
+	"time"
+
+	"github.com/miekg/dns"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	NoSuchHost = errors.New("no such host")
+)
+
+func IsNoSuchHostError(err error) bool {
+	return err.Error() == NoSuchHost.Error()
+}
+
+type HostResolver interface {
+	LookupIPAddr(ctx context.Context, host string) ([]HostAddress, error)
+}
+
+type HostAddress struct {
+	Host string
+	IP   gonet.IP
+	TTL  time.Duration
+	TXT  string
+}
+
+// ConfigMapHostResolver is a HostResolver that looks up the IP address of
+// a host from a ConfigMap. Used for testing purposes
```

Comment:
as mentioned above. I don't think we need the configmap resolver yet? So I'd prefer to remove it, and implement something neater when we require it.

Reply:
do we know what something is? I worry we will just end up adding this back when we want to add more e2e tests etc?

Reply:
I don't want to put the task of implementing a new e2e dns resolver on this unrelated PR - so I'm avoiding discussing how we could make a better one here. I'm just concerned that we're adding this when we don't need it. Which makes it more likely we'll go the way of reusing it, when we get to writing e2e tests. Why add something that isn't working very well for us, when we don't even need it yet?

In summary, I'm not looking for you to write or create a new mock dns resolver, just delete this unrequired and suboptimal copy/pasted code.

Reply:
sure ok will update